docs/contributing-to-piwik-core.md

```diff
@@ -160,6 +160,49 @@ Here are best practises we aim to follow when creating, reviewing and merging pu
 * Read the article: [Pull Requests: How to Get and Give Good Feedback](https://www.kickstarter.com/backing-and-hacking/pull-requests-how-to-get-and-give-good-feedback)
 * When a pull request has many small commits, it is recommended to [Squash commits](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit) so that there is only one (bigger) commit.
 
+## Switching between branches
+
+### Keeping dependencies up to date
+
+Every time you change the branch you may need to run composer install (and maybe also npm install). You can automate this task by following these steps:
```

Comment:
It's pretty rare that this needs to be done, isn't it? We only update when needed to (security fixes or major version), so this might make switching branches slower (I don't use it so I'm not sure).

Reply:
It's rare it needs to actually install a newer version but when it's needed then it can be hard to find out why it's not working or harder to reproduce things since you're not testing correct version. And when there's nothing to update then it executes fairly fast as it only executes when the file actually changes (so only when needed).